File: src/lib-ui/measure-text.js
//~ function measure_text(x)
//~ {
 //~ check(is_str,x)

 //~ const context=canvas.getContext("2d") 
 //~ const o=context.measureText(x) 
 //~ const width=trunc(o.width)
 //~ const height=add(o.fontBoundingBoxAscent,o.fontBoundingBoxDescent)
 
 //~ return {width,height}
//~ }

function measure_text(x)
{
 check(is_str,x)
 
 const o=document.createElement("span")
 
 o.innerHTML=x
 o.style.font=font
 
 document.body.appendChild(o)
 
 const width=o.offsetWidth
 const height=o.offsetHeight
 
 o.remove()
 
 return {width,height}
}

File: src/prj-core/lib-ui/measure-text.cs
fn measure_text x:str
 let span dom_span

 dom_html span x

 h_font_family span font_family
 h_font_size span font_size

 dom_push body span

 let width span.offsetWidth
 let height span.offsetHeight

 span.remove

 ret obj width height
end
